Cython/Compiler/PyrexTypes.py

```diff
@@ -4448,7 +4469,8 @@ def create_to_py_utility_code(self, env):
                     "module_name": module_name,
                     "is_flag": True,
                     },
-            outer_module_scope=self.entry.scope  # ensure that "name" is findable
+            outer_module_scope=self.entry.scope,  # ensure that "name" is findable
+            compiler_directives = directives,
```

Comment:
> Does this need to be done here, or can the SwitchTransform do it?

I wondered about this. It'd essentially mean that the switch transform wouldn't be applied to (most?) extern enums. Right now we apply it but document it as a possible cause of invalid C code. So it would be removing an optimization in some cases.

I was trying to do the thing with the smallest behaviour change, but either option is ok

Reply:
Cython assuming extern enums can be used with the switch statement had been a problem for me, and I'm currently using `#cython: optimize.use_switch=False` in my projects.

Reply:
For now I've deduplicated the code but not changed the behaviour of `SwitchTransform`.

I think the way to do it would be to make `optimize.use_switch=False` still turn on `SwitchTransform`, but only in  cases where we absolutely know it's unconditionally safe (i.e. no unknown values). Then in the enum utility code, we'd always turn off `optimize.use_switch` and let the transform decide whether it's safe. That feels like a larger change that's probably outside this PR.